validation/validator.go

```diff
@@ -30,11 +38,39 @@ type failedValidator struct {
 	err error
 }
 
+// IsStringInSlice tells if a searched value is part of a slice or not
+func IsStringInSlice(slice []string, searched string) bool {
+	for _, value := range slice {
+		if value == searched {
+			return true
+		}
+	}
+	return false
+}
+
 // NewParameterValidator creates a validator ready to check multiple parameters
 func NewParameterValidator() Validator {
 	return &successValidator{}
 }
 
+func (v *successValidator) ValidateParameter(prmName string, validable Validable, mandatory bool) Validator {
+	if validable == nil {
+		if mandatory {
+			return &failedValidator{err: cerrors.CreateMissingParameterError(prmName)}
+		}
+	} else {
+		return v.ValidateParameterFunc(prmName, validable.Validate)
+	}
+	return v
+}
+
+func (v *successValidator) ValidateParameterFunc(prmName string, validator func() error) Validator {
```

Comment:
why do we need prmName if it's not used in the method?

Reply:
Parameter removed